### PR TITLE
Add integerations for socket and grpc

### PR DESCRIPTION
**Pull request by hossein-raeisi**
- The [gRPC](https://grpc.io/) integration instruments all incoming requests and outgoing unary-unary, unary-stream grpc requests using grpcio channels.
Use this integration to start or continue transactions for incoming grpc requests, create spans for outgoing requests,
and ensure traces are properly propagated to downstream services.
- The Socket integration to create spans for dns resolves and connection creations.

Docs: https://github.com/getsentry/sentry-docs/pull/6327

### CODE REVIEW

**Comment by Hossain-raeisi** [comment]
docs [pull request](https://github.com/getsentry/sentry-docs/pull/6327)

**Comment by Hossain-raeisi** [comment]
I think this merge request has reached a stable state, would appreciate a review @sl0thentr0py  @antonpirker

**Comment by sl0thentr0py** [comment]
ty so much for the contribution @Hossain-raeisi , we'll try to get to this next week

**Comment by antonpirker** [comment]
Hey @Hossain-raeisi 
Amazing contribution! As @sl0thentr0py said we will have a look next week, we are already very stoked about this!

**Comment by Hossain-raeisi** [comment]
Hey!
Any updates @sl0thentr0py @antonpirker ?

**Comment by antonpirker** [comment]
Have not had time yet. Please hang in there.

**Comment by antonpirker** [comment]
Hey @Hossain-raeisi !
At first glance it looks good.
I fixed some smaller things to make CI happy. 

Still, there are some errors in CI mostly about typing:
https://github.com/getsentry/sentry-python/actions/runs/4321738769/jobs/7543323597

Could you fix them?
(Ignore the gevent errors, I have to fix those)

**Comment by Hossain-raeisi** [comment]
Hi @antonpirker 
thanks for the review 
I made some changes and hopefully they should solve most of the errors, but I dont think I can do anything about these ones 
https://github.com/getsentry/sentry-python/actions/runs/4321738769/jobs/7543323597#step:4:81

**Comment by Hossain-raeisi** [comment]
Hey @antonpirker!
do you have any free time in next few days to pair and finish up this pr?

**Review by antonpirker — CHANGES_REQUESTED** [review]

File: sentry_sdk/integrations/grpc/server.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        # type: (ServerInterceptor, Callable[[HandlerCallDetails], RpcMethodHandler], HandlerCallDetails) -> RpcMethodHandler
+        handler = continuation(handler_call_details)
+        if not handler or not handler.unary_unary:
+            return handler
+
+        def behavior(request, context):
+            # type: (Message, ServicerContext) -> Message
+            hub = Hub(Hub.current)
+
+            name = self._find_method_name(context)
```
[the remark this answers is not in the document]
Comment (antonpirker) [reply sentry_sdk/integrations/grpc/server.py:35]: I have now added a small fix to your PR concerning the `_find_method_name` function. @Oogwey can you check if this is what you wanted to do?

**Comment by antonpirker** [comment]
I created a small demo project and the traces are propagated perfectly! Really cool!

Screenshot 2023-03-28 at 17 11 33

**Review by antonpirker — APPROVED** [review]

**Comment by antonpirker** [comment]
I guess when you @Oogwey are ok with the changes that I did in your PR, then we can merge this!

**Comment by antonpirker** [comment]
oh, and we have to fix the typing errors. I will look at this tomorrow!

**Comment by Oogwey** [comment]
> I guess when you @Oogwey are ok with the changes that I did in your PR, then we can merge this!

changes seem alright to me, thanks

**Comment by antonpirker** [comment]
I changed now the setup, that GRPC tests are only run in Python 3.7+ (because the protobuffers in the tests did not work with version installed on older pythons)

Is this fine with you @Oogwey ?

**Comment by Oogwey** [comment]
> I changed now the setup, that GRPC tests are only run in Python 3.7+ (because the protobuffers in the tests did not work with version installed on older pythons)
> 
> Is this fine with you @Oogwey ?
alright with me :+1:

**Comment by Oogwey** [comment]
> I have one more ask @Oogwey , could you please rename `TestService`, `TestMessage` and `test_grpc_server` and prefix it with something like `Grpc` or whatever. Pytest it trying to collect tests from all files prefixed `test_` and all classes prefixed `Test` so pytest struggles with out code. Thank You!

Done!

**Comment by antonpirker** [comment]
I am not sure why the Python 2.7 tests are suddenly failing, because those are tests you and me did not touch. 🤷 Very strange. (thought it is because of a file called `socket.py`, but there are other things also failing...

Could you please merge the newest `master` from `getsentry/sentry-python` into your branch (or rebase yours on top of it?) Maybe this helps. Thanks @hossein-raeisi

**Comment by hossein-raeisi** [comment]
> I am not sure why the Python 2.7 tests are suddenly failing, because those are tests you and me did not touch. shrug Very strange. (thought it is because of a file called `socket.py`, but there are other things also failing...
> 
> Could you please merge the newest `master` from `getsentry/sentry-python` into your branch (or rebase yours on top of it?) Maybe this helps. Thanks @hossein-raeisi


strange :thinking: 
seems like is already merged:
image
(remote sentry refers to getsentry/sentry-python)

**Comment by antonpirker** [comment]
Hm. Now all the tests pass. I guess it was a GitHub hiccup, because they had problems with Github actions in the last couple of days.

**Comment by hossein-raeisi** [comment]
> Congrats and thanks for you patience @hossein-raeisi !
> 
> Really great contribution! Thank you so much! We will release this in the next couple of days.
> 
> If you would like a little thank you gift please send me your shipping address and t-shirt size to [anton.pirker@sentry.io](mailto:anton.pirker@sentry.io)

thanks, I would love that

**Comment by antonpirker** [comment]
Your contributions have been released @hossein-raeisi : https://github.com/getsentry/sentry-python/releases/tag/1.19.0

**Pull request merged**